lib/routes/route_stats.ex

```diff
@@ -107,6 +107,7 @@ defmodule Routes.RouteStats do
   end
 
   @spec median_schedule_adherence(t()) :: number() | nil
+  @spec median_schedule_adherence(stats_by_route(), String.t()) :: number() | nil
```

Comment:
```suggestion
  @spec median_schedule_adherence(stats_by_route(), Route.id()) :: number() | nil
```
thought: Certainly not necessary, but you could also go with `Route.id()` if you wanted to hint more at the data coming in.